tests/templates/kuttl/smoke-nodeport/02-validate-all-ingresses-are-reachable.yaml

```
@@ -33,23 +40,33 @@ kind: Job
 metadata:
   name: query-ingresses
 spec:
+  backoffLimit: 1
   template:
     spec:
-      restartPolicy: OnFailure
+      restartPolicy: Never
       serviceAccountName: query-ingresses-sa
       containers:
         - name: query-ingresses
           image: docker.stackable.tech/stackable/testing-tools:0.2.0-stackable0.0.0-dev
           command:
             - sh
-            - -c
+            - -euc
             - |
-              for pod in nginx-0 nginx-1; do \
+              echo Testing access via Listener.status
+              for pod in nginx-0 nginx-1; do
                 ADDR=$(kubectl get listener/listener-$pod \
-                  -o jsonpath='http://{.status.ingressAddresses[0].address}:{.status.ingressAddresses[0].ports.http}/pod-name' \
-                  -n $NAMESPACE); \
-                echo Requesting $ADDR should return $pod; \
-                curl -s $ADDR | grep $pod; \
+                  -o jsonpath='http://{.status.ingressAddresses[0].address}:{.status.ingressAddresses[0].ports.http}/pod-name');
+                echo Requesting $ADDR should return $pod;
+                curl -s $ADDR | grep $pod;
+              done
+
+              echo Testing access via PodListeners
+              for pod in nginx-0 nginx-1; do
+                POD_UID=$(kubectl get pod/$pod -o jsonpath='{.metadata.uid}' -n $NAMESPACE)
```

The namespace is not required here and can be removed:

```suggestion
                POD_UID=$(kubectl get pod/$pod -o jsonpath='{.metadata.uid}')
```

The `env` structure can then also be removed.

---

True.

---

e96f248eac9e237e36a73fa44a0cc5e7094bdc0b